perfkitbenchmarker/benchmarks/mysql_service_benchmark.py

```diff
@@ -0,0 +1,773 @@
+                        vm.db_instance_name,
+                        '--password', MYSQL_ROOT_PASSWORD]
+    stdout, stderr, _ = vm_util.IssueCommand(set_password_cmd)
+    logging.info('Set root password completed. Stdout:\n%s\nStderr:\n%s',
+                 stdout, stderr)
+
+  def Cleanup(self, vm):
+    if hasattr(vm, 'db_instance_name'):
+      delete_db_cmd = [FLAGS.gcloud_path,
+                       'sql',
+                       'instances',
+                       'delete', vm.db_instance_name,
+                       '--quiet']
+
+      stdout, stderr, status = vm_util.IssueCommand(delete_db_cmd)
+      logging.info('DB cleanup command issued, stdout is %s, stderr is %s '
+                   'status is %s', stdout, stderr, status)
+    else:
+      logging.info('db_instance_name does not exist, no need to cleanup.')
+
+
+MYSQL_SERVICE_BENCHMARK_DICTIONARY = {
+    benchmark_spec_class.GCP: GoogleCloudSQLBenchmark(),
+    benchmark_spec_class.AWS: RDSMySQLBenchmark()}
+
+
+def Prepare(benchmark_spec):
+  """Prepare the MySQL DB Instances, configures it.
+     Prepare the client test VM, installs SysBench, configures it.
+
+  Args:
+    benchmark_spec: The benchmark specification. Contains all data that is
+        required to run the benchmark.
+  """
+  # We would like to always cleanup server side states.
+  # If we don't set this, our cleanup function will only be called when the VM
+  # is static VM, but we have server side states to cleanup regardless of the
+  # VM type.
+
+  benchmark_spec.always_call_cleanup = True
+
+  vms = benchmark_spec.vms
+
+  # Setup common test tools required on the client VM
+  vms[0].Install('sysbench05plus')
+
+  # Prepare service specific states (create DB instance, configure it, etc)
+  MYSQL_SERVICE_BENCHMARK_DICTIONARY[FLAGS.cloud].Prepare(vms[0])
+
+
+def Run(benchmark_spec):
+  """Run the MySQL Service benchmark and publish results.
+
+  Args:
+    benchmark_spec: The benchmark specification. Contains all data that is
+        required to run the benchmark.
+
+  Returns:
+    Results.
+  """
+  logging.info('Start benchmarking MySQL Service, '
+               'Cloud Provider is %s.', FLAGS.cloud)
+  vms = benchmark_spec.vms
+  metadata = {}
+  metadata['oltp_tables_count'] = FLAGS.oltp_tables_count
+  metadata['oltp_table_size'] = FLAGS.oltp_table_size
+  metadata['db_instance_cores'] = FLAGS.db_instance_cores
+  metadata['sysbench_warm_up_seconds'] = FLAGS.sysbench_warmup_seconds
+  metadata['sysbench_run_seconds'] = FLAGS.sysbench_run_seconds
+  metadata['sysbench_thread_count'] = FLAGS.sysbench_thread_count
+  metadata['sysbench_latency_percentile'] = FLAGS.sysbench_latency_percentile
+  metadata['sysbench_report_interval'] = FLAGS.sysbench_report_interval
+
+  # The run phase is common across providers. The VMs[0] object contains all
+  # information and states necessary to carry out the run.
+  results = _RunSysbench(vms[0], metadata)
+  print results
```

Comment:
I intentionally used "print" here because it allows me to write the final set of results to stdout in addition to stderr. (The caller, pkb.py, logs the results to stderr using logging.info()). 

The benefit of printing these to stdout is it gives me a short and concise view of the result (rather than mixing those up with all the info logs), and it's valuable for me when I need to do bulk processing/grepping results from lots of different runs. (All I need to do is to look at the stdout files which are much shorter and faster to read than stderr files of many runs.)

Some trick I learned when doing the object storage benchmarks and automations. :-)